de.dlr.sc.virsat.model.extension.fdir.test/src/de/dlr/sc/virsat/model/extension/fdir/recovery/ParallelComposerTest.java

```diff
@@ -0,0 +1,238 @@
+/*******************************************************************************
+ * Copyright (c) 2008-2019 German Aerospace Center (DLR), Simulation and Software Technology, Germany.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package de.dlr.sc.virsat.model.extension.fdir.recovery;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import de.dlr.sc.virsat.model.dvlm.concepts.Concept;
+import de.dlr.sc.virsat.model.extension.fdir.model.RecoveryAutomaton;
+import de.dlr.sc.virsat.model.extension.fdir.model.State;
+import de.dlr.sc.virsat.model.extension.fdir.model.Transition;
+import de.dlr.sc.virsat.model.extension.fdir.util.RecoveryAutomatonHelper;
+import de.dlr.sc.virsat.model.extension.fdir.util.RecoveryAutomatonHolder;
+
+/**
+ * Class to test the ParallelComposer
+ * @author jord_ad
+ *
+ */
+public class ParallelComposerTest {
+	
+	protected Concept concept;
+	protected ParallelComposer pc;
+	
+	@Before
+	public void setUp() throws Exception {
+		String conceptXmiPluginPath = "de.dlr.sc.virsat.model.extension.fdir/concept/concept.xmi";
+		concept = de.dlr.sc.virsat.concept.unittest.util.ConceptXmiLoader.loadConceptFromPlugin(conceptXmiPluginPath);
+		
+		pc = new ParallelComposer();
+	}
+	
+	@Test
+	public void testNull() {
+		RecoveryAutomaton result = pc.compose(null);
+		assertNull(result);
+	}
+	
+	@Test
+	public void testNone() {
+		RecoveryAutomaton result = pc.compose(new HashSet<RecoveryAutomaton>());
+		assertNull(result);

```

Comment:
In would expect the empty recovery automaton rather than 0 when calling compose on an empty set.